File: ios/Reguerta/Reguerta/Domain/Shifts/ShiftPlanningRequest.swift
import Foundation

nonisolated enum ShiftPlanningRequestType: String, Equatable, Sendable {
    case delivery
    case market
}

nonisolated enum ShiftPlanningRequestStatus: String, Equatable, Sendable {
    case requested
    case processing
    case completed
    case failed
}

nonisolated struct ShiftPlanningPreviewReference: Equatable {
    let sourceRequestId: String
    let bundleRevision: String
    let bundleDigest: String
}

nonisolated enum ShiftPlanningRequestIntent: Equatable {
    case preview
    case stage(ShiftPlanningPreviewReference)
}

nonisolated struct ShiftPlanningRequest: Identifiable, Equatable {
    let id: String
    let bundleId: String
    let requestedByUserId: String
    let requestedAtMillis: Int64
    let deliveryTargetSeasonStartYear: Int
    let marketTargetSeasonStartYear: Int
    let intent: ShiftPlanningRequestIntent
}

File: android/Reguerta/app/src/main/java/com/reguerta/user/domain/shifts/ShiftPlanningRequest.kt
package com.reguerta.user.domain.shifts

enum class ShiftPlanningRequestType {
    DELIVERY,
    MARKET,
}

enum class ShiftPlanningRequestStatus {
    REQUESTED,
    PROCESSING,
    COMPLETED,
    FAILED,
}

data class ShiftPlanningRequest(
    val id: String,
    val bundleId: String,
    val requestedByUserId: String,
    val requestedAtMillis: Long,
    val deliveryTargetSeasonStartYear: Int,
    val marketTargetSeasonStartYear: Int,
)
